program_transformers/src/lib.rs

```diff
@@ -254,3 +255,27 @@ fn record_metric(metric_name: &str, success: bool, retries: u32) {
         cadence_macros::statsd_count!(metric_name, 1, "success" => success, "retry_count" => retry_count);
     }
 }
+
+pub async fn skip_metadata_json_download<T>(
+    download_metadata_info: &DownloadMetadataInfo,
+    conn: &T,
+) -> bool
+where
+    T: ConnectionTrait + TransactionTrait,
+{
+    let DownloadMetadataInfo {
+        asset_data_id,
+        slot: incoming_slot,
+        uri,
+    } = download_metadata_info;
+
+    asset_data::Entity::find_by_id(asset_data_id.clone())
+        .one(conn)
+        .await
+        .unwrap_or(None)
+        .is_some_and(|model| {
+            !model.reindex.unwrap_or(false)
+                && model.metadata_url.eq(uri)
+                && model.slot_updated >= *incoming_slot
+        })
+}
```

Comment:
I think the das-core crate metdata_json.rs is the spot for this function. Also recommend keeping it private if we can and just expose download_notifier.